### ISSUE TITLE
Mix between Yarn and npm

### ISSUE THREAD

**Issue by Tmarty** [chromaui/learnstorybook.com#153]
I just completed the tutorial and enjoyed learning about storybook.

However I noticed a mix of `npm` and `yarn` commands throughout the tutorial. Would you be interested in a PR consolidating the commands to use either `npm` or `yarn` throughout for consistency? And if so, which?

**Comment by domyen** [comment]
@tmeasday what do you think?

**Comment by jonniebigodes** [comment]
@Tmarty you've beat to the punch, i was compiling a list of fixes so that the documentation could move forward. But in this case you addressed it. From my read i think that npm would be the choice, for the following reason, newcomers to the ecosystem will be prompted to use npm as the de facto package manager, more experienced ones will know by then that yarn is a option and when they follow the documentation using yarn. But that's just my two cents i'll defer to @domyen and @tmeasday on how to proceed.

**Comment by tmeasday** [comment]
I kind of agree @jonniebigodes although one other factor is that `create-react-app` uses `yarn` (and creates a `yarn.lock`), and `sb init` also follows that. So we kind of accidentally use `yarn` in the first couple of steps.

I'm not sure what happens if the user doesn't have `yarn` installed at that point.

**Comment by jonniebigodes** [comment]
one test i'm ok with doing is, remove yarn from my system, leaving in only npm as the de facto package manager  and purge any cached information and try it with only npm. I can do it tomorrow as it's almost 2 am where i'm at and i want to finish the translation of the introduction on the "design systems for developers" document. Do you mind waiting a bit @tmeasday and @Tmarty ?

**Comment by tmeasday** [comment]
Of course! I can try too sometime soon

**Comment by jonniebigodes** [comment]
Sorry for the delay, but the testing took a bit more than expected.
I'm going to enumerate the steps i took for this issue below:

- Started by purging the yarn cache, it took me a bit more than expected as you can see
![test_yarn_1](https://user-images.githubusercontent.com/22988955/66276535-13fe2900-e88c-11e9-8b4b-b7783e46af67.png)
- Also purged the npm cache to get a clean slate
![test_yarn_2](https://user-images.githubusercontent.com/22988955/66276544-237d7200-e88c-11e9-9723-a0d6eb186d51.png)
- As you can see i'm using windows, so i went ahead and uninstalled yarn as you would normally do with any windows app.
- Ran `yarn version` and it reported that a version 1.2.1. was still present, dig a bit deeper into npm cache folders, which is something of a fun ride(pardon the bad pun) and checked that it was still lingering there. Ran `npm uninstall -g yarn` to remove it and ran again `yarn version` and i'm now presented with:
![test_yarn_3](https://user-images.githubusercontent.com/22988955/66276583-90910780-e88c-11e9-9033-3e3dae33d3ae.png)
- Re-installed `create react app´ globally
![test_yarn_4](https://user-images.githubusercontent.com/22988955/66276620-dc43b100-e88c-11e9-8f00-eb6ab18f26bc.png)
- Opened the storybook tutorial and started following the steps, issued `npx create-react-app taskbox` and after a bit i'm presented with
![test_yarn_5](https://user-images.githubusercontent.com/22988955/66276658-19a83e80-e88d-11e9-9389-18775cc2d6f0.png)
- Checked for the presence of a `yarn.lock` file and i'm presented with:
![test_yarn_6](https://user-images.githubusercontent.com/22988955/66276669-3c3a5780-e88d-11e9-8250-674192288f5f.png)
- Issued `npx -p @storybook/cli sb init`, to add storybook to the website and initialize it and i'm presented with the following:
![test_yarn_7](https://user-images.githubusercontent.com/22988955/66276752-73a90400-e88d-11e9-9182-e9eb42fb05f9.png)

and 

![test_yarn_8](https://user-images.githubusercontent.com/22988955/66276763-87546a80-e88d-11e9-807a-4790b18d8a71.png)

- Checked again for any presence of yarn, and no mentions as you can see:
![test_yarn_9](https://user-images.githubusercontent.com/22988955/66276772-ab17b080-e88d-11e9-8362-af420bcfd547.png)

- Continued following the tutorial, issued ``npm run test` it yelds the following:
![test_yarn_10](https://user-images.githubusercontent.com/22988955/66276792-e4e8b700-e88d-11e9-895f-0d62402bacd6.png)
- Following the tutorial, issued `npm run storybook` and i'm presented with:
![test_yarn_11](https://user-images.githubusercontent.com/22988955/66276812-0e094780-e88e-11e9-9179-798f2e05101b.png)
- And finally `npm run start` and it yelds the following:
![test_yarn_12](https://user-images.githubusercontent.com/22988955/66276821-237e7180-e88e-11e9-87a3-9b9794485ffe.png)

 For brevity purposes i'm not showing any more screenshots, but i ran the whole tutorial all with npm only and it all checked out. Waiting to see what @tmeasday's tests offer. But so far it seems that all works fine with npm. But i'm inclined for this small change in the documentation, move the snippets to npm and removing yarn as it would allow consistency across the documentation and also for the reasons mentioned in my earlier comment.
 Probably in the [get started](https://www.learnstorybook.com/intro-to-storybook/react/en/get-started/) add something along the lines of "alternatively instead of npm, you can use yarn as your package manager, as the tutorial works with either". Now, something worthwile mentioning is that i've only covered the react tutorial, i haven't checked with either angular or vue, but i'm inclined to believe that it shouldn't present any issues. I'll defer to @domyen and @tmeasday for input.

**Comment by tmeasday** [comment]
Thanks for the detailed investigation @jonniebigodes.

My only remaining concern is inexperienced users who happen to have both `yarn` and `npm` installed (i.e. they've installed `yarn` at some stage for whatever reason, but aren't experienced enough to mentally translate between `npm` and `yarn` commands).

The issue for such users is that `create-react-app` will default to `yarn` but all subsequent commands will use `npm`.  However, we can fix that by changing the first call to:

```bash
npx create-react-app taskbox --use-npm
```

I checked and if you do that:
  a) CRA uses `npm` and creates a `package-lock.json`
  b) then, `sb init` *also* uses `npm` and continues.

So my proposal would be

a) add the `--use-npm` flag to the first command
b) add a note at that point saying "if you want to use `yarn`, drop the `--use-npm` and replace all `npm` commands with the equivalent `yarn` commands"
c) change all calls to use `npm`.

Sound good everyone?

**Comment by jonniebigodes** [comment]
@tmeasday i fully understand your reasoning, i was only focusing on experienced users and new users relative to the ecosystem. And wow i didn't know that, the flag option, wow someone learned something new today(pardon the bad pun). And your proposal looks like the perfect fit for each case. I'm up for it as it will streamline the whole documentation.

**Comment by Tmarty** [comment]
Great investigation and I learned something new too :)

Let me know if you want any help implementing this.

**Comment by ElixirMike** [comment]
I have followed the steps above, but when I run "npm run test", I see "Error: no test specified".
What am I missing?

**Comment by jonniebigodes** [comment]
@ElixirMike if you don't mind i'm going to take a look at this and report back, i need to confirm something before giving a more detailed answer. Do you mind waiting a bit?

**Comment by ElixirMike** [comment]
I figured it out. I uninstalled yarn, and was running into an issue where the create-react-app was not correctly installing . I had to npm install create-react-app again, and then everything worked.

**Comment by jonniebigodes** [comment]
@ElixirMike i wouldn't go to the extreme of removing yarn. Your issue kept me thinking and i took this for another "spin" (pardon the bad pun) and actually found something interesting. I followed @tmeasday 's suggestion and the Storybook documentation, more specifically the [get started](https://www.learnstorybook.com/intro-to-storybook/react/en/get-started/) for your scenario and another scenario that is related to a pr currently under review, running `npm run test` or `yarn test` i'm presented with the following:

![elixirmike_1](https://user-images.githubusercontent.com/22988955/74354502-5e039800-4db3-11ea-8cf4-0e29b2eafe08.png)

the cli ommits it, but on the left side is the result of running `npm run test` and on the right side the result of running ``yarn test` and the message clearly states:

```bash
No tests found related to files changed since last commit.
Press `a` to run all tests, or run Jest with `--watchAll`.
```
I was thinking that some recent change introduced into the cra package might have removed the tests, i checked the boilerplate code and i'm presented with the following:

![elixirmike_2](https://user-images.githubusercontent.com/22988955/74354957-0b76ab80-4db4-11ea-8f44-2cd6736ae9e9.png)

Continued onto the next section of the tutorial and you're introduced to a type of testing, more specifically snapshot testing. Upon completing the steps and running `npm run test` i'm presented with the following:

![elixirmike_3](https://user-images.githubusercontent.com/22988955/74356405-2fd38780-4db6-11ea-942a-553934f48c68.png)

The test runs and the message clears away. This led me to believe that the until you add something in the means of a test file with some code, cra will not not run the tests, or more specifically will not run the test until you manually trigger it by pressing the `a` key to force it or you commit some changes to the original test file setup when installing or add a new one. 

Going over this and despite the mention in the cli, this is a bit of a regression on their part. Taking this from a perspective of a person that never touched jest or React or a newcomer to the ecosystem this might be misleading, a better approach would be to run the original test no matter what and let the developer see some type of result and go from there. As the developer starts to get his/hers "testing legs" then it's up to him/her to act upon.


Anyway this is just my 2 cents, feel free to provide some feedback.

**Comment by tmeasday** [comment]
@jonniebigodes So perhaps we should pass the `--all` to the command we tell people to run initially to check tests are working? I think somewhere along the line CRA started defaulting to watch mode, which as you've noticed only runs tests that have changed in the current commit (so none initially).

**Comment by jonniebigodes** [comment]
@tmeasday that was my thought aswell and to prevent a situation like the one @ElixirMike mentioned it would be a best bet to apply the change, i can take care of it during the weekend as i have to work on the react pt translation as is a bit outdated.

**Comment by jonniebigodes** [comment]
Fixed in #283
